workflow/controller/dag.go

```diff
@@ -117,6 +117,12 @@ func (d *dagContext) getTaskNode(taskName string) *wfv1.NodeStatus {
 	return &node
 }
 
+// isExpandedNode returns true if this node is a result of a task expansion (i.e. withParam, withItems, etc.)
+// nodeName should be the name of the node, not the task
+func isExpandedNode(nodeName string) bool {
+	return strings.Contains("(", nodeName)
+}
```

Comment:
Will write exhaustive tests for this function if we decide to pursue this approach